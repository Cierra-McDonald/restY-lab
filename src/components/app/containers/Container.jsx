import React, { Component } from 'react'
import Controls from './Controls'

export default class Container extends Component {
    state = { 
        url: '',
        rawJSON: '', 
    }

    handleURLChange = (e) => { 
        e.preventDefault();
        console.log(e.target.value)
        this.setState({ 
            url: e.target.value
        })
    }

    handleSubmitChange = () => { 


    }

    render() {
        return (
            <div>
              <Controls
              onChange={this.handleURLChange}
              />  
            </div>
        )
    }
}
