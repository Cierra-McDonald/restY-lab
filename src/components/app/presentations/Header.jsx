import React from 'react'
import styles from './Presentations.module.css'


function Header() {

    return (
        <div className={styles.divheader}>
            <h1 className={styles.appheader}>REST-less</h1>
        </div>
    )
}


export default Header;
