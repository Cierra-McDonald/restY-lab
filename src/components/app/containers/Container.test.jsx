import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import Container from './Container';
import Header from '../presentations/Header'

describe('testing for Container presentation', () => {
  it('renders the container page with all of its components', async() => {
    render(<Header/>);

	screen.getByText('REST-less');

    render(<Container/>)

    const ul = screen.getByRole('list', { name: 'history-list' });
    expect(ul).toMatchSnapshot();

    const span = screen.getByLabelText('json-results');
    expect(span).toMatchSnapshot();

  });
});