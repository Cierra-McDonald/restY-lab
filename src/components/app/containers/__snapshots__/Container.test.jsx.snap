// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for Container presentation renders the container page with all of its components 1`] = `
<ul
  aria-label="history-list"
>
  <li>
    <div>
      GET
       
      http://imthebomb.com
    </div>
  </li>
  <li>
    <div>
      POST
       
      http://imthebomb.com
    </div>
  </li>
  <li>
    <div>
      PUT
       
      http://imthebomb.com
    </div>
  </li>
  <li>
    <div>
      DELETE
       
      http://imthebomb.com
    </div>
  </li>
</ul>
`;

exports[`testing for Container presentation renders the container page with all of its components 2`] = `
<div
  aria-label="json-results"
>
  <h3>
    Results!
  </h3>
  <p>
    "[{fake: but real to me}]"
  </p>
</div>
`;
